Code.Rmd

```diff
@@ -29,23 +42,126 @@ Congratulations!
+Another file you often see in the wild is `R/utils.R`.
+This is a common place to define small utilities that are used inside multiple package functions.
+Since they serve as helpers to multiple functions, placing them in `R/utils.R` makes them easier to re-discover when you return to your package after a long break.
+
+*TODO: link to Bob Rudis's blog post? <https://rud.is/b/2018/04/08/dissecting-r-package-utility-belts/>*
+:::
+
+If it's very hard to predict which file a function lives in, that suggests its time to separate your functions into more files or reconsider how you are naming your functions and/or files.
+
+:::rstudio-tip
+The arrangement of functions within files is less important in RStudio, which offers two ways to jump to the definition of a function:
+
+* Press Ctrl + . (the period) then start typing the name. Keep typing to narrow
+  the list and eventually pick a function (or file) to visit. This works for both
+  functions and files in your project.
+* With your cursor in a function name or with a function name selected, press
+  F2. This works for functions defined in your package or in another package.
+  
+After navigating to a function with one of these methods, return to where you started by clicking the back arrow at the top-left of the editor or by pressing Ctrl + F9 (Windows & Linux) or Cmd + F9 (macOS).
+:::
 
-*removed in deference to material in <https://style.tidyverse.org>; see [tidyverse/style/#121](https://github.com/tidyverse/style/issues/121)*
+*TODO: I've let two images go here because unclear how that would/should work inside an RStudio tip. Are we OK with this? See them in the 1st ed at the end of this section: <http://r-pkgs.had.co.nz/r.html#r-organising>*
 
 ## Code style
 
-*removed in deference to material in <https://style.tidyverse.org>; see [tidyverse/style/#122](https://github.com/tidyverse/style/issues/122)*
+We recommend following the tidyverse style guide (<https://style.tidyverse.org>), which goes into much more detail than we can here.
+Its format also allows it to be a more dynamic document than this book.
+
+Although the style guide explains the "what" and the "why", another important decision is *how* to enforce a specific code style.
+For this we recommend the styler package (<https://styler.r-lib.org>); its default behaviour enforces the tidyverse style guide.
+There are many ways to apply styler to your code, depending on the context:
+
+* `styler::style_pkg()` restyles an entire R package.
+* `styler::style_dir()` restyles all files in a directory.
+* `usethis::use_tidy_style()` is wrapper that applies one of the above functions
+  depending on whether the current project is an R package or not.
+* `styler::style_file()` restyles a single file.
+* `styler::style_text()` restyles a character vector.
+
+:::rstudio-tip
+When styler is installed, the RStudio Addins menu will offer several additional ways to style code:
+
+* the active selection
+* the active file
+* the active package
+:::
```

Comment:
Mention rebinding RStudio's default re-style keyboard shortcut? (that's what I do)